compiler/hash-typecheck/src/traverse.rs

```diff
@@ -1032,35 +1109,75 @@ impl<'c, 'w, 'm, 'g, 'i> visitor::AstVisitor<'c> for ModuleTypechecker<'c, 'w, '
         ctx: &Self::Ctx,
         node: ast::AstNodeRef<ast::LetStatement<'c>>,
     ) -> Result<Self::LetStatementRet, Self::Error> {
```

Comment:
Maybe split this in two functions to handle bounds separately?

Reply:
Yeah, I will do this once I also add support for trait bounds.